### DISCUSSION TITLE
Training on Imbalanced Data

### DISCUSSION THREAD

**Discussion by mlnomadpy** [tensorflow/decision-forests#189] [General]
Hello,
Is there any implementation of class weighting in TensorFlow decision forests, similar to class_weight in Keras models?

**Comment by rstz** [comment]
Hi,
the preferred option is to set weights manually (e.g. give an explicit weight to each example). Doing so is possible by either adding a third channel to the tf.Dataset fed to TF-DF (see [here](https://github.com/tensorflow/decision-forests/blob/7ffa80ac0b4647ac3f7afc302bb685745985cc13/tensorflow_decision_forests/keras/keras_test.py#L619)) or (when converting from pandas) by specifying the `weight` parameter in `pd_dataframe_to_tf_dataset`, see [here](https://github.com/tensorflow/decision-forests/blob/7ffa80ac0b4647ac3f7afc302bb685745985cc13/tensorflow_decision_forests/keras/keras_test.py#L647). 

For binary classification, you can also set the `class_weight` parameter of the `fit` function. However, this is not supported for distributed training. 

Finally the underlying C++ library Yggdrasil Decision Forests (YDF) provides more flexibility for defining weights through [this proto message](https://github.com/google/yggdrasil-decision-forests/blob/main/yggdrasil_decision_forests/dataset/weight.proto) linked in the [training configuration](https://github.com/google/yggdrasil-decision-forests/blob/main/yggdrasil_decision_forests/learner/abstract_learner.proto#L112), but this is currently not fully exposed in TF-DF

**Reply by mlnomadpy** [reply]
Thank you very much for your response,
I will try to try the `pd_dataframe_to_tf_dataset` approach.
